tests/Integration/Behaviour/Features/Context/Domain/Discount/DiscountFeatureContext.php

```diff
@@ -268,6 +305,18 @@ protected function createDiscount(string $cartRuleReference, array $data, AddDis
             }
         }
 
+        if ($command instanceof AddFreeGiftDiscountCommand) {
+            if (!empty($data['gift_product'])) {
+                $reference = $data['gift_product'];
+                $productId = $this->getProductIdByReference($reference);
```

Comment:
No need for this method `getProductIdByReference` we already have a generic `referenceToId` method for that